apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -65,22 +76,27 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
                   )
                   .map(([name]) => name);
 
-    useEffect(() => {
-        const newSearchParams = new URLSearchParams(searchParams);
-        newSearchParams.set('module', selectedModule);
-        setSearchParams(newSearchParams);
-    }, [selectedModule, setSearchParams, searchParams]);
-
     const submitSearch = useCallback(() => {
-        setSelectedModule((prev: string) =>
-            filteredModules.length === 1 ? filteredModules[0] : prev
-        );
-    }, [filteredModules]);
+        if (filteredModules.length === 1) {
+            const convertedSearchParams = new URLSearchParams(searchParams);
+            convertedSearchParams.set('module', filteredModules[0]);
+            setSearchParams(convertedSearchParams);
+        }
+    }, [filteredModules, setSearchParams, searchParams]);
+
+    const onChangeModule = useCallback(
+        (newModule: string) => () => {
+            const convertedSearchParams = new URLSearchParams(searchParams);
+            convertedSearchParams.set('module', newModule);
+            setSearchParams(convertedSearchParams);
+        },
+        [setSearchParams, searchParams]
+    );
 
     return (
         <div className="flex flex-col md:flex-row md:flex-nowrap gap-5 border-0 border-y border-solid border-sui-grey-45">
             <div className="w-full md:w-1/5">
-                <Combobox value={selectedModule} onChange={setSelectedModule}>
+                <Combobox value={selectedModule} onChange={onChangeModule}>
```

Comment:
This `onChange` doesn't look like it'd work with the current API, although if you make the change I suggest above it wouldn't be an issue.